pkg/collector/corechecks/agentprofiling/agentprofiling.go

```diff
@@ -192,6 +196,48 @@ func (m *Check) Run() error {
 	return nil
 }
 
+// terminateAgent requests graceful shutdown of the agent process after flare generation completes.
+// It uses the agent's established shutdown mechanism (signals.Stopper) which ensures proper cleanup
+// via stopAgent(). Termination is skipped when running in test mode to avoid killing the test process.
+func (m *Check) terminateAgent() {
+	// Skip termination when running in test mode
+	// Check if we're running under go test by looking for test-related arguments
```

Comment:
Recommend using [testing.Testing()](https://pkg.go.dev/testing#Testing) or build tagging (`//go:build !test`) instead of this somewhat fragile logic.